chain/client/src/sync/state/shard.rs

```diff
@@ -45,11 +45,6 @@ impl Drop for StateSyncShardHandle {
     }
 }
 
-/// The maximum parallelism to use per shard. This is mostly for fairness, because
-/// the actual rate limiting is done by the TaskTrackers, but this is useful for
-/// balancing the shards a little.
```

Comment:
Do you know why we introduced this shard balancing in the first place? If we want to keep it, should the buffered below use something like limit/num of shards we're syncing?

Reply:
I've gone ahead and removed the functional changes. Added a new configuration variable for per-shard limits that defaults to the same value as this constant.